crates/ironrdp-rdpdr/src/pdu/efs.rs

```diff
@@ -968,57 +970,60 @@ impl ServerDeviceAnnounceResponse {
-    /// STATUS_ACCESS_DENIED
-    AccessDenied = 0xC000_0022,
-    /// STATUS_NOT_A_DIRECTORY
-    NotADirectory = 0xC000_0103,
-    /// STATUS_NO_SUCH_FILE
-    NoSuchFile = 0xC000_000F,
-    /// STATUS_NOT_SUPPORTED
-    NotSupported = 0xC000_00BB,
-    /// STATUS_DIRECTORY_NOT_EMPTY
-    DirectoryNotEmpty = 0xC000_0101,
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct NtStatus(u32);
+
+impl NtStatus {
+    pub const SUCCESS: Self = Self(0x0000_0000);
+    pub const UNSUCCESSFUL: Self = Self(0xC000_0001);
+    pub const NOT_IMPLEMENTED: Self = Self(0xC000_0002);
+    pub const NO_MORE_FILES: Self = Self(0x8000_0006);
+    pub const OBJECT_NAME_COLLISION: Self = Self(0xC000_0035);
+    pub const ACCESS_DENIED: Self = Self(0xC000_0022);
+    pub const NOT_A_DIRECTORY: Self = Self(0xC000_0103);
+    pub const NO_SUCH_FILE: Self = Self(0xC000_000F);
+    pub const NOT_SUPPORTED: Self = Self(0xC000_00BB);
+    pub const DIRECTORY_NOT_EMPTY: Self = Self(0xC000_0101);
 }
 
 impl TryFrom<u32> for NtStatus {
     type Error = PduError;
 
     fn try_from(value: u32) -> Result<Self, Self::Error> {
         match value {
-            0x0000_0000 => Ok(NtStatus::Success),
-            0xC000_0001 => Ok(NtStatus::Unsuccessful),
-            0xC000_0002 => Ok(NtStatus::NotImplemented),
-            0x8000_0006 => Ok(NtStatus::NoMoreFiles),
-            0xC000_0035 => Ok(NtStatus::ObjectNameCollision),
-            0xC000_0022 => Ok(NtStatus::AccessDenied),
-            0xC000_0103 => Ok(NtStatus::NotADirectory),
-            0xC000_000F => Ok(NtStatus::NoSuchFile),
-            0xC000_00BB => Ok(NtStatus::NotSupported),
-            0xC000_0101 => Ok(NtStatus::DirectoryNotEmpty),
-            _ => Err(invalid_message_err!("try_from", "NtStatus", "unsupported value")),
+            0x0000_0000 => Ok(NtStatus::SUCCESS),
+            0xC000_0001 => Ok(NtStatus::UNSUCCESSFUL),
+            0xC000_0002 => Ok(NtStatus::NOT_IMPLEMENTED),
+            0x8000_0006 => Ok(NtStatus::NO_MORE_FILES),
+            0xC000_0035 => Ok(NtStatus::OBJECT_NAME_COLLISION),
+            0xC000_0022 => Ok(NtStatus::ACCESS_DENIED),
+            0xC000_0103 => Ok(NtStatus::NOT_A_DIRECTORY),
+            0xC000_000F => Ok(NtStatus::NO_SUCH_FILE),
+            0xC000_00BB => Ok(NtStatus::NOT_SUPPORTED),
+            0xC000_0101 => Ok(NtStatus::DIRECTORY_NOT_EMPTY),
+            _ => Err(custom_err!("NtStatus", UnsupportedNtStatus(value))),
         }
     }
 }
 
 impl From<NtStatus> for u32 {
     fn from(status: NtStatus) -> Self {
-        status as u32
+        status.0
     }
 }
 
+/// An error type for when we receive an NTSTATUS value that we don't (yet) support.
+#[derive(Debug)]
+pub struct UnsupportedNtStatus(u32);
+
+impl std::fmt::Display for UnsupportedNtStatus {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:#010X}", self.0)
+    }
+}
+
+impl std::error::Error for UnsupportedNtStatus {}
+
```

Comment:
Do we actually need to restrict the values we put into `NtStatus`?

Reply:
Ah yes, now I understand. No, we have no need for restricting these.

998151bcbd39e274296e0f7799f4e5896b661fa2